fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanSchedularServiceImpl.java

```diff
@@ -184,4 +191,138 @@ public void recalculateInterest() throws JobExecutionException {
 
 	}
 
+	@Override
+	@CronTarget(jobName = JobName.RECALCULATE_INTEREST_FOR_LOAN)
+	public void recalculateInterest(Map<String, String> jobParameters) {
+		//gets the officeId
+		final String officeId = jobParameters.get("officeId");
+		logger.info(officeId);
+		Long officeIdLong=Long.valueOf(officeId);
+		//gets the Office object
+		final Office office = this.officeRepository.findOne(officeIdLong);
+		if(office == null)
+			throw new OfficeNotFoundException(officeIdLong);
+		final int threadPoolSize=Integer.parseInt(jobParameters.get("thread-pool-size"));
+		final int batchSize=Integer.parseInt(jobParameters.get("batch-size"));
+
+		recalculateInterest(office,threadPoolSize,batchSize);
+	}
+
+	@Override
+	public void recalculateInterest(Office office, int threadPoolSize, int batchSize) {
+		final int pageSize = batchSize * threadPoolSize;
+
+		//initialise the executor service with fetched configurations
+		final ExecutorService executorService = Executors.newFixedThreadPool(threadPoolSize);
+
+		Long maxLoanIdInList = 0L;
+		final String officeHierarchy = office.getHierarchy() + "%";
+
+		//Get the loanIds from service
+		List<Long> loanIds = this.loanReadPlatformService
+				.fetchLoansForInterestRecalculation(pageSize, maxLoanIdInList, officeHierarchy);
+
+		// gets the loanIds data set iteratively and call addAccuruals for that paginated dataset
+		do {
+			int totalFilteredRecords = loanIds.size();
+			logger.info("Starting accrual - total filtered records - " + totalFilteredRecords);
+			recalculateInterest(loanIds, threadPoolSize, batchSize,
+					executorService);
+//            if (totalFilteredRecords > 0) {
```

Comment:
did you mean to leave this commented out code in here?  It's usually best not to introduce new commented code in PRs... if it's needed, uncomment it, if it's not, then just delete it.